crates/rpc/rpc-eth-api/src/helpers/transaction.rs

```diff
@@ -643,3 +646,58 @@ pub trait LoadTransaction: SpawnBlocking {
         }
     }
 }
+
+/// Find a transaction by sender and nonce
+pub async fn get_transaction_by_sender_and_nonce<Eth>(
+    eth: &Eth,
+    sender: Address,
+    nonce: u64,
+    include_pending: bool,
+) -> Result<Option<RpcTransaction<Eth::NetworkTypes>>, Eth::Error>
+where
+    Eth: EthBlocks + EthState,
+{
+    // Check the pool first
+    if include_pending {
+        if let Some(tx) = LoadState::pool(eth).get_transaction_by_sender_and_nonce(sender, nonce) {
+            let transaction = tx.transaction.clone().into_consensus();
+            return Ok(Some(from_recovered(transaction)));
+        }
+    }
+
+    // Check if the sender is a contract
+    if EthState::get_code(eth, sender, None).await?.len() > 0 {
+        return Ok(None);
+    }
+
+    let highest = EthState::transaction_count(eth, sender, None).await?.saturating_to::<u64>();
```

Comment:
add transaction count to `LoadState` (see other comment for bg)
```suggestion
    let highest = LoadState::transaction_count(eth, sender, None).await?.saturating_to::<u64>();
```